tracing-subscriber/src/registry/sharded.rs

```diff
@@ -622,7 +620,7 @@ mod tests {
         // registry.
         let dispatch = dispatcher::Dispatch::new(subscriber);
 
-        let span2 = dispatcher::with_default(&dispatch, || {
+        let span2 = dispatcher::with_default(dispatch, || {
```

Comment:
similarly to my above comment, this needs to be
```suggestion
        let span2 = dispatcher::with_default(dispatch.clone(), || {
```